thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/cache/ThirdEyeCacheResponse.java

```diff
@@ -127,26 +127,29 @@ public boolean isMissingEndSlice(long sliceEnd) {
    */
   private void checkAndLogMissingMiddleSlices() {
 
-    List<String> missingTimestamps = new ArrayList<>();
-    long timeGranularity = request.getRequest().getGroupByTimeGranularity().toMillis();
-
-    // remember that we return the cached timeseries in sorted order,
-    // but this assumption is not necessarily true if mergeSliceIntoRows() has been called.
-    for (int i = 1; i < timeSeriesRows.size(); i++) {
-      long previousTimestamp = timeSeriesRows.get(i - 1).getTimestamp();
-      long currentTimestamp = timeSeriesRows.get(i).getTimestamp();
-
-      // add all missing timestamps between previous timestamp and current timestamp to
-      // the list of missing timestamps.
-      while (previousTimestamp + timeGranularity < currentTimestamp) {
-        missingTimestamps.add(String.valueOf(previousTimestamp + timeGranularity));
-        previousTimestamp += timeGranularity;
+    if (!this.hasNoRows()) {
```

Comment:
call it isEmpty instead of hasNoRows. Much simpler to read.

Reply:
done